### ISSUE TITLE
Peak Heating and Cooling Load Occurring on Different Days in First Month Cause Crash

### ISSUE THREAD

**Issue by mitchute** [j-c-cook/ghedt#79]
See the following load profiles which are very similar. `Clg-0pct.csv` has zero cooling load in the first month, while `Clg-10pct` has a relatively small cooling load in the first month. A plot of the first and last day of the year for these loads is shown below.

![Screen Shot 2022-01-19 at 4 04 51 PM](https://user-images.githubusercontent.com/2985979/150232552-c6ae7b38-26fc-4f74-9862-2026cb42d309.png)

GHEDT will compute a size for `Clg-0pct.csv`; however, it will crash for `Clg-10pct`. Tracing the issue through the code - initially, I found that the natural log of a negative value was attempting to be compted here:

https://github.com/j-c-cook/ghedt/blob/2a1238eac13d0c86f64eee937fccf305feed31f0/ghedt/ground_heat_exchangers.py#L168

However, continuing to trace back to the origin of the issue, we find that negative first hours for the peak heating and cooling loads are being computed here:

https://github.com/j-c-cook/ghedt/blob/2a1238eac13d0c86f64eee937fccf305feed31f0/ghedt/peak_load_analysis_tool/ground_loads.py#L610-L621

I suspect that this is due to issues related to adding the last day of the year on to the beginning of the simulation for cases when the peak load occurs at the beginning of the simulaiton. Perhaps a workaround is to just make sure that these peak hours never go negative?

GHEDT SHA: 2a1238eac13d0c86f64eee937fccf305feed31f0

Load Profiles
[Clg-0pct.csv](https://github.com/j-c-cook/ghedt/files/7901044/Clg-0pct.csv)
[Clg-10pct.csv](https://github.com/j-c-cook/ghedt/files/7901045/Clg-10pct.csv)

**Comment by j-c-cook** [comment]
Initially upon running `Clg-10pct.csv` in the [near_square.py](https://github.com/j-c-cook/ghedt/blob/main/ghedt/examples/Design/find_near_square.py) design example file, I got the following message:
```
ValueError: Based on the loads provided, the excess temperatures for the minimum and maximum number of boreholes falls above 0. This means that the loads are "astronomical" or that the higher end of the domain needs to contain more boreholes. Consider increasing the available land area, or decreasing the minimum allowable borehole spacing.
```
This error was correct. A new issue needs to address the fact that the `Design` interface has a hard coded 1x1 to 32x32 range for the `near_square` search in the following lines:
https://github.com/j-c-cook/ghedt/blob/2a1238eac13d0c86f64eee937fccf305feed31f0/ghedt/design.py#L39-L42

After modifying the `32` to be `64` I could get the program to throw the error @mitchute mentions (He may be using a routine that allows him to define a land area, or perhaps he did the same thing using the near-square search). A new issue will need to address this hard-coded value. 

The issue here is that there is not peak heating load (`self.monthly_peak_hl_day[i] == 0`). Then `self.monthly_peak_hl_day[i] - 1 == -1`. This only has the possibility of occurring for the first month. It happens several times with these loads throughout the design process. 

For a temporary fix to make the program run through, I modified the lines highlighted here: https://github.com/j-c-cook/ghedt/pull/80#discussion_r788271984. 

Utilizing the `find_near_square.py` example file, the resulting system is output. 
The result of the fix is as follows:
```
Calculation time: 121.56 seconds
Height: 132.8260 meters
Number of boreholes: 1369
Total Drilling: 181838.8 meters
```
This is a `37 x 37` square field. 

To verify that it is okay to change the values to be near zero like Matt mentioned, I ran the following check with the detailed hourly simulation. The bug here is in the peak load analysis (PLAT). A way to simulate the system without using the PLAT is a hourly simulation. These take a while. I've added the following lines to the program:
```
# Size the system with an hourly simulation
    ghe.size(method='hourly')

    subtitle = '* Single U-tube'
    print(subtitle + '\n' + len(subtitle) * '-')
    print('Calculation time: {0:.2f} seconds'.format(toc - tic))
    print('Height: {0:.4f} meters'.format(ghe.bhe.b.H))
    nbh = len(ghe.GFunction.bore_locations)
    print('Number of boreholes: {}'.format(nbh))
    print('Total Drilling: {0:.1f} meters\n'.
          format(ghe.bhe.b.H * nbh))
```
This likely will take all night to run given the size of the system.

**Comment by j-c-cook** [comment]
One change to the `find_near_square.py` example is that I set borehole spacing to be `B=10.` meters. The example file here currently has `B=5.` meters.

**Comment by j-c-cook** [comment]
Geese nevermind this thing really cranks. 
```
Calculation time: 123.82 seconds
Height: 135.0000 meters
Number of boreholes: 1369
Total Drilling: 184815.0 meters
```
I'll fix and merge this. Going to get #81 done tomorrow and then release a hot-fix `Version 0.1.1`.

**Comment by mitchute** [comment]
Thanks, @j-c-cook. I've confirmed the fix addresses the issue.
